cli/azd/pkg/rzip/rzip.go

```diff
@@ -6,55 +6,129 @@ package rzip
 import (
 	"archive/zip"
 	"io"
-	"io/fs"
+	"log"
 	"os"
 	"path/filepath"
 	"strings"
 )
 
+// CreateFromDirectory creates a zip archive from the given directory recursively,
+// that is suitable for transporting across machines.
+//
+// It resolves any symlinks it encounters.
 func CreateFromDirectory(source string, buf *os.File) error {
 	w := zip.NewWriter(buf)
-	err := filepath.WalkDir(source, func(path string, info fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
 
-		if info.IsDir() {
-			return nil
-		}
-		fileInfo, err := info.Info()
-		if err != nil {
-			return err
-		}
+	err := addDirRoot(w, source)
+	if err != nil {
+		return err
+	}
 
-		header := &zip.FileHeader{
-			Name: strings.Replace(
-				strings.TrimPrefix(
-					strings.TrimPrefix(path, source),
-					string(filepath.Separator)), "\\", "/", -1),
-			Modified: fileInfo.ModTime(),
-			Method:   zip.Deflate,
-		}
+	return w.Close()
+}
 
-		f, err := w.CreateHeader(header)
+func addDirRoot(
+	w *zip.Writer,
+	src string) error {
+	return addDir(w, "", src, 0)
+}
```

Comment:
nit: I don't think this would violate our line length rule - any reason for wrapping this function signature?

Reply:
nothing in particular, just keeping it similar in style with other functions here